### PR TITLE
feature/telegram-chats-dropdown

### PR DESCRIPTION
**Pull request by Sorizen**

### CODE REVIEW

File: src/common/modal/RefreshModal.vue [hunk 1]
```diff
@@ -177,6 +179,12 @@ const isSubmoduleStepRefreshCanBeShown = computed(
     Boolean(submodules.value.length),
 )
 
+const submoduleName = computed(() =>
```
Comment (WhiteNik16) [inline src/common/modal/RefreshModal.vue]: I think we can use destructuring
`const submoduleName = computed(() => {
  const { value } = submoduleToSearch;
  return value.charAt(0) === '/' ? value.slice(1) : value;
});`

File: src/common/modal/RefreshModal.vue [hunk 2]
```diff
@@ -249,15 +264,23 @@ const checkIsSubmoduleLinkValid = async () => {
...
 
 const searchSubmodule = async () => {
+  if (!submoduleName.value) return
+  const isSubmoduleWithThisNameExist = submodules.value.some(
+    item => item === submoduleName.value,
+  )
+  if (isSubmoduleWithThisNameExist) {
+    isSuchSubmoduleExistError.value = true
+    return
+  }
```
Comment (WhiteNik16) [inline src/common/modal/RefreshModal.vue:267]: I think we can use destructuring, as in the example above

File: src/fields/InputDropdownField.vue [hunk 3]
```diff
@@ -104,51 +109,27 @@ const closeDropdown = () => {
...
-  await loadUsers()
+const selectCurrentUser = (currentUser: InputDropdownPickOption) => {
+  toggleDropdown()
+  emit('update:modelValue', currentUser.id)
+  emit('update:searchValue', currentUser.text)
 }
 
-const debounceSearchUsers = debounce(search, 200)
-
 onMounted(async () => {
```
Comment (WhiteNik16) [inline src/fields/InputDropdownField.vue]: remove async property

File: src/forms/PermissionForm.vue [hunk 4]
```diff
@@ -248,6 +270,10 @@ const isLoadingPermissionsError = ref(false)
 const loadingPermissionsResponseErrorCode = ref<HTTP_STATUS_CODES | ''>('')
 const isAccessLevelRequired = ref(false)
 const accessList = ref<ModulePermissions[]>([])
+const telegramChats = ref<TelegramChat[]>([])
+const isTelegramChatsLoaded = ref(false)
+const isTelegramChatsLoadingError = ref(false)
+const currentTelegramChatId = ref(NaN)
```
Comment (WhiteNik16) [inline src/forms/PermissionForm.vue]: Are you sure you should use NaN here

File: src/forms/VerifyUserForm.vue [hunk 5]
```diff
@@ -86,7 +96,10 @@ const emit = defineEmits<{
 
 const { $t } = useContext()
 const { currentUserId } = useAuthStore()
-const selectedUser = ref<VerifiedUser | null>(null)
+const users = ref<VerifiedUser[]>([])
+const selectedUserId = ref(NaN)
```
Comment (WhiteNik16) [inline src/forms/VerifyUserForm.vue]: Are you sure you should use NaN here

**Review by WhiteNik16 — CHANGES_REQUESTED** [review]

File: src/common/modal/RefreshModal.vue (hunk 1 above)
Comment (Sorizen) [reply src/common/modal/RefreshModal.vue]: 1. It's not consistent
2. This is an unnecessary complication of logic
3. Decomposition is not used in such cases.

File: src/common/modal/RefreshModal.vue (hunk 2 above)
Comment (Sorizen) [reply src/common/modal/RefreshModal.vue:267]: 1. It's not consistent
2. This is an unnecessary complication of logic
3. Decomposition is not used in such cases.
4. This code will be with error. When you decompose a Proxy object, you get the value of the variable, but if you change it, this field in the Proxy will not change. You should change value from Proxy by setter.

File: src/fields/InputDropdownField.vue (hunk 3 above)
Comment (Sorizen) [reply src/fields/InputDropdownField.vue]: done

File: src/forms/PermissionForm.vue (hunk 4 above)
Comment (Sorizen) [reply src/forms/PermissionForm.vue]: Done

File: src/forms/VerifyUserForm.vue (hunk 5 above)
Comment (Sorizen) [reply src/forms/VerifyUserForm.vue]: Done

File: src/common/modal/RefreshModal.vue [hunk 6]
```diff
@@ -177,6 +179,12 @@ const isSubmoduleStepRefreshCanBeShown = computed(
     Boolean(submodules.value.length),
 )
 
+const submoduleName = computed(() =>
+  submoduleToSearch.value.charAt(0) === '/'
+    ? submoduleToSearch.value.slice(1)
+    : submoduleToSearch.value,
```
Comment (KlausKidman) [inline src/common/modal/RefreshModal.vue]:
```
submoduleToSearch.value.replace(/^\//, '')

File: src/fields/InputDropdownField.vue [hunk 7]
```diff
@@ -104,51 +110,27 @@ const closeDropdown = () => {
...
-  closeDropdown()
+const search = (searchValue: string) => {
+  emit('update:searchValue', searchValue)
 }
 
-const search = async () => {
-  users.value = []
-  isDropdownOpened.value = true
-  await loadUsers()
+const selectCurrentUser = (currentUser: InputDropdownPickOption) => {
```
Comment (KlausKidman) [inline src/fields/InputDropdownField.vue]: rename this function

File: src/fields/InputDropdownField.vue [hunk 8]
```diff
@@ -43,58 +67,40 @@
...
+
+type SCHEMES = 'default' | 'secondary'
 
-const props = withDefaults(
+withDefaults(
   defineProps<{
-    modelValue: string
-    user: VerifiedUser | null
-    label?: string
+    searchValue: string
```
Comment (KlausKidman) [inline src/fields/InputDropdownField.vue]: update logic with one outgoing value

File: src/fields/InputDropdownField.vue [hunk 9]
```diff
@@ -43,58 +67,40 @@
...
+    placeholder: '',
+    dropdownScheme: 'default',
+    modelValue: undefined,
   },
 )
 
 const emit = defineEmits<{
-  (e: 'update:user', value: VerifiedUser | null): void
-  (e: 'update:modelValue', value: string): void
+  (e: 'update:modelValue', value: number): void
```
Comment (KlausKidman) [inline src/fields/InputDropdownField.vue]: i think `modelValue` might be `string` or `number`

File: src/forms/PermissionForm.vue [hunk 10]
```diff
@@ -260,6 +286,10 @@ const isUsernameInputDisabled = computed(() => Boolean(form.phoneNumber))
 
 const isPhoneInputDisabled = computed(() => Boolean(form.username))
 
+const parsedUsername = computed(() =>
+  form.username.charAt(0) === '@' ? form.username.slice(1) : form.username,
```
Comment (KlausKidman) [inline src/forms/PermissionForm.vue]:
```
form.username.replace(/^\@/, '')

File: src/forms/PermissionForm.vue [hunk 11]
```diff
@@ -360,13 +410,32 @@ const submit = async () => {
...
+                    : null,
+                }
+              : {}),
+            ...(isTelegramModule.value
+              ? {
+                  submodule_id: selectedTelegramChat.value
+                    ? selectedTelegramChat.value.attributes.submodule_id
+                    : null,
+                }
+              : {}),
```
Comment (KlausKidman) [inline src/forms/PermissionForm.vue]: join to one `isTelegramModule.value`

File: src/forms/PermissionForm.vue [hunk 12]
```diff
@@ -360,13 +410,32 @@ const submit = async () => {
...
             link: form.link,
             access_level: accessLevelValue?.value,
-            ...(form.username ? { username: form.username } : {}),
+            ...(form.username
+              ? {
+                  username: parsedUsername.value,
+                }
+              : {}),
             ...(form.phoneNumber
               ? { phone: prefix.value + form.phoneNumber.split(' ').join('') }
```
Comment (KlausKidman) [inline src/forms/PermissionForm.vue]:
```
phone: prefix.value + form.phoneNumber.replaceAll(' ', '')

File: src/forms/PermissionForm.vue [hunk 13]
```diff
@@ -416,10 +486,28 @@ const getAccessLevelList = async () => {
...
         filter: {
           link: form.link,
-          ...(form.username ? { username: form.username } : {}),
+          ...(form.username
+            ? {
+                username: parsedUsername.value,
+              }
+            : {}),
           ...(form.phoneNumber
             ? { phone: prefix.value + form.phoneNumber.split(' ').join('') }
```
Comment (KlausKidman) [inline src/forms/PermissionForm.vue]:
```
phone: prefix.value + form.phoneNumber.replaceAll(' ', '')

File: src/forms/PermissionForm.vue [hunk 14]
```diff
@@ -416,10 +486,28 @@ const getAccessLevelList = async () => {
...
+                  : null,
+              }
+            : {}),
+          ...(isTelegramModule.value
+            ? {
+                submodule_id: selectedTelegramChat.value
+                  ? selectedTelegramChat.value.attributes.submodule_id
+                  : null,
+              }
+            : {}),
```
Comment (KlausKidman) [inline src/forms/PermissionForm.vue]: join to one isTelegramModule.value

File: src/pages/user-details/ModuleTreesItem.vue [hunk 15]
```diff
@@ -217,6 +217,12 @@ const deleteUserFromModule = async () => {
             user_id: String(props.id),
             link: props.item.link,
             username: props.item.username,
+            ...(isTelegramModule.value
+              ? { submodule_access_hash: props.item.submodule_access_hash }
+              : {}),
+            ...(isTelegramModule.value
+              ? { submodule_id: props.item.submodule_id }
+              : {}),
```
Comment (KlausKidman) [inline src/pages/user-details/ModuleTreesItem.vue]: to join

File: src/common/modal/RefreshModal.vue (hunk 6 above)
Comment (Sorizen) [reply src/common/modal/RefreshModal.vue]: done

File: src/fields/InputDropdownField.vue (hunk 7 above)
Comment (Sorizen) [reply src/fields/InputDropdownField.vue]: done

File: src/forms/PermissionForm.vue (hunk 10 above)
Comment (Sorizen) [reply src/forms/PermissionForm.vue]: done

File: src/fields/InputDropdownField.vue (hunk 8 above)
Comment (Sorizen) [reply src/fields/InputDropdownField.vue]: done

File: src/fields/InputDropdownField.vue (hunk 9 above)
Comment (Sorizen) [reply src/fields/InputDropdownField.vue]: done

File: src/forms/PermissionForm.vue (hunk 12 above)
Comment (Sorizen) [reply src/forms/PermissionForm.vue]: done

File: src/forms/PermissionForm.vue (hunk 13 above)
Comment (Sorizen) [reply src/forms/PermissionForm.vue]: done

File: src/forms/PermissionForm.vue (hunk 11 above)
Comment (Sorizen) [reply src/forms/PermissionForm.vue]: doene everywhere

File: src/forms/PermissionForm.vue (hunk 14 above)
Comment (Sorizen) [reply src/forms/PermissionForm.vue]: done

File: src/pages/user-details/ModuleTreesItem.vue (hunk 15 above)
Comment (Sorizen) [reply src/pages/user-details/ModuleTreesItem.vue]: done

**Review by KlausKidman — APPROVED** [review]

**Pull request merged**